_posts/2025-09-30-Introducing-Derived-Source.markdown

```diff
@@ -0,0 +1,143 @@
+---
+layout: post
+title:  "Save up to 2x on storage with derived source"
+authors:
+   - tanikp
+   - mgodwani
+   - bukhtawa
+   - kolchfa
+date: 2025-09-30
+categories:
+  - technical-posts
+meta_keywords: todo
+meta_description: "Learn about derived source, introduced in OpenSearch 3.2: why it matters, how it works, and how to start using it to reduce storage costs."
+---
+
+Storage is the key factor when deciding on the optimal configuration of OpenSearch cluster hence key factor in driving infrastructure cost of your OpenSearch cluster. As your data grows, storage requirement can increase in multifold factor, given how OpenSearch stores document in multiple formats. Derived source feature is here to optimise storage cost.
+
+In this blog post, we will describe how documents are stored in OpenSearch and how to use derived source to retrieve those documents without compromising on most of the legacy functionality and with significant storage reduction.
+
+## How documents are stored in OpenSearch?
+
+When documents are ingested, OpenSearch stores the original document body in the [`_source`](https://docs.opensearch.org/latest/field-types/metadata-fields/source/) fields and also document’s fields get stored in various forms like [indexed](https://docs.opensearch.org/latest/field-types/mapping-parameters/index-parameter/), [stored](https://docs.opensearch.org/latest/field-types/mapping-parameters/store/) and [docValues](https://docs.opensearch.org/latest/field-types/mapping-parameters/doc-values/). OpenSearch stores data in different format as each field type requires values to be stored in specific form for optimised search, e.g. full-text search relies on inverted index whereas exact term aggregation on keyword field relies on doc values. As original document gets stored in a separate data-structure comprising all the fields at a single place, it becomes easy to fetch while doing the query rather that fetching individual fields from the various disc locations. With this setup, search latency benefit is there at the expense of storage cost due to duplication of data.
```

Comment:
> As original document gets stored in a separate data-structure comprising all the fields at a single place, it becomes easy to fetch while doing the query rather that fetching individual fields from the various disc locations. With this setup, search latency benefit is there at the expense of storage cost due to duplication of data.

These lines aren't easy to comprehend. Maybe try using an example query to demonstrate how different formats are retrieved with source getting used in the fetch phase only.